### ISSUE TITLE
[New Resource]: `aws_dms_data_migration` 

### ISSUE THREAD

**Issue by jar-b** [hashicorp/terraform-provider-aws#39684]
### Description

AWS Database Migration Service recently released new CRUD API's around data migrations. A corresponding resource would enable practitioners to manage data migration resources via Terraform.

### Requested Resource(s) and/or Data Source(s)

- `aws_dms_data_migration`

### Potential Terraform Configuration

```terraform
resource "aws_dms_data_migration" "example" {
  data_migration_type          = ""
  migration_project_identifier = ""
  service_access_role_arn      = ""
  data_migration_name          = ""
  
  source_data_settings {
    # settings
  }
}
```

### References

- depends on #39679 
- https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/databasemigrationservice#Client.CreateDataMigration
- https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/databasemigrationservice#Client.DescribeDataMigrations
- https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/databasemigrationservice#Client.ModifyDataMigration
- https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/databasemigrationservice#Client.DeleteDataMigration

### Would you like to implement a fix?

None

**Comment by gavinclarkeuk** [comment]
Duplicate of https://github.com/hashicorp/terraform-provider-aws/issues/38004?
